test/org/sagebionetworks/bridge/play/controllers/ConsentControllerMockedTest.java

```diff
@@ -19,36 +31,115 @@
         controller.setConsentService(consentService);
 
-        ParticipantOptionsService optionsService = mock(ParticipantOptionsService.class);
+        optionsService = mock(ParticipantOptionsService.class);
         controller.setOptionsService(optionsService);
 
         controller.setCacheProvider(mock(CacheProvider.class));
+    }
 
+    @Test
+    public void testChangeSharingScope() {
         controller.changeSharingScope(SharingScope.NO_SHARING, "message");
+
         InOrder inOrder = inOrder(optionsService, consentService);
         inOrder.verify(optionsService).setOption(study, "healthCode", SharingScope.NO_SHARING);
         inOrder.verify(consentService).emailConsentAgreement(study, user);
     }
+
+    @Test
+    public void consentSignatureJSONCorrect() throws Exception {
+        ConsentSignature sig = new ConsentSignature.Builder().withName("Jack Aubrey").withBirthdate("1970-10-10")
+                .withImageData("data:asdf").withImageMimeType("image/png").withSignedOn(UNIX_TIMESTAMP).build();
+
+        when(consentService.getConsentSignature(study, user)).thenReturn(sig);
+
+        Result result = controller.getConsentSignature();
+
+        String json = Helpers.contentAsString(result);
+        JsonNode node = BridgeObjectMapper.get().readTree(json);
+
+        assertEquals(5, fieldNameCount(node));
+        assertEquals("Jack Aubrey", node.get("name").asText());
+        assertEquals("1970-10-10", node.get("birthdate").asText());
+        assertEquals("ConsentSignature", node.get("type").asText());
+        assertEquals("data:asdf", node.get("imageData").asText());
+        assertEquals("image/png", node.get("imageMimeType").asText());
+        // no signedOn value when serializing
+    }
+    
+    @Test
+    public void consentSignatureHasSignedOnValue() throws Exception {
+        // This signedOn value should be ignored, it is always set on the server
+        String json = "{\"name\":\"Jack Aubrey\",\"birthdate\":\"1970-10-10\",\"signedOn\":123,\"imageData\":\"data:asdf\",\"imageMimeType\":\"image/png\",\"scope\":\"no_sharing\"}";
+        
+        Context context = TestUtils.mockPlayContextWithJson(json);
+        Http.Context.current.set(context);
+        
+        ArgumentCaptor<ConsentSignature> captor = ArgumentCaptor.forClass(ConsentSignature.class);
+        when(consentService.consentToResearch(any(Study.class), any(User.class), captor.capture(),
+                any(SharingScope.class), any(Boolean.class))).thenReturn(user);
+        
+        Result result = controller.giveV2();
+        String response = Helpers.contentAsString(result);
+        JsonNode node = BridgeObjectMapper.get().readTree(response);
+        assertEquals("Consent to research has been recorded.", node.get("message").asText());
+        
+        ConsentSignature signature = captor.getValue();
+        // This will be much higher than 123L
+        assertTrue(signature.getSignedOn() > DateTime.now().minusMinutes(1).getMillis());
```

Comment:
Would it be easier to use DateTimeUtils.setCurrentMillisFixed() so we know exactly when "now" is?